nextest-runner/src/lib.rs

```diff
@@ -24,6 +24,7 @@ pub mod redact;
 pub mod reporter;
 pub mod reuse_build;
 pub mod runner;
+mod rustc_cli;
```

Comment:
@sunshowers This crate(`nextest-runner`) probably is not the best place to add the new `rustc_cli` module which streamlines the `rustc --print target-libdir` command call, but:

* the current design of the `BuildPlatformBuilder` in the `cargo-nextest` requires both the host libdir and the target libdir string as an input; and
* the `runner_for_target` function in the `nextest-runner/tests/integration/target_runner.rs` file needs correct libdir strings to set up tests correctly

so we need to find a way to call `RustcCli` in both `cargo-nextest` and `nextest-runner`, and we end up with this solution. The alternatives are:

1. Do not set up or hard coded host libdir and target libdir for the `nextest-runner` integration tests, so that `nextest-runner` doesn't need `RustcCli`, and we can move it to the `cargo-nextest` crate; or
2. We can call `RustcCli` when we build `BuildPlatform`, so that the integration test of the `nextest-runner` crate doesn't explicitly depend on `RustcCli`, and we don't have to export it.
    cons:
        * we still need `RustcCli` to live in the `nextest-runner` crate
        * to make testing `BuildPlatform` easier I need to still keep the current interface that accepts string, which complicates the interface
3. Move `RustCli` to `cargo-nextest` and add a `dev-dependencies` entry to `nextest-runner` to depend on `cargo-nextest`.
    cons:
        * The dependency relationship becomes strange(may result in cyclic)
        *  We still need to expose `RustcCli` as a public interface